subsys/nfc/ndef/le_oob_rec_parser.c

```diff
@@ -0,0 +1,369 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <string.h>
+#include <errno.h>
+#include <logging/log.h>
+#include <sys/byteorder.h>
+#include <nfc/ndef/le_oob_rec_parser.h>
+#include <nfc/ndef/payload_type_common.h>
+
+LOG_MODULE_REGISTER(nfc_ndef_le_oob_rec_parser,
+		    CONFIG_NFC_NDEF_LE_OOB_REC_PARSER_LOG_LEVEL);
+
+/* Size of AD fields. */
+#define AD_TYPE_FIELD_SIZE 1UL
+#define AD_LEN_FIELD_SIZE 1UL
+
+/* Incorrect Value of AD Structure Length field. */
+#define AD_LEN_INC_VAL  1UL
+
+static u8_t *memory_allocate(u8_t **buf, u32_t *buf_len, u32_t alloc_size)
```

Comment:
No, I haven't thought about it.

I took a look at the ```net_buf_simple``` API now, but I would rather use dedicated function for allocating data so that this module works similarly to our generic NDEF parser.

Error handling is based on asserts in ```net_buf_simple``` API (when memory cannot be allocated). However, we return error codes in our parser libraries (ENOMEM).

Reply:
The functions used for error handling `net_buf_tailroom` `net_buf_headroom` are public functions and can be used to return error codes instead.

Re-using net_buf would be an advantage in maintainability, amount of code that must be tested. and code size.

Reply:
I used ```net_buf_simple``` as a memory allocator. Please review if this is what you had in mind while suggesting this change.

I added this change as a separate commit so it's easy for you to see ```net_buf_simple``` related changes.